File: frontend/src/types/errorType.ts
import type { PageData } from './api'

export type ErrorType = {
  id: number
  parentId: number | null
  typeLevel: 1 | 2
  code: string
  name: string
  description: string | null
  sortOrder: number
  enabled: boolean
  createdAt: string
  updatedAt: string
}

export type ErrorTypeLevelFilter = '' | '1' | '2'
export type ErrorTypeEnabledFilter = '' | 'true' | 'false'

export type ErrorTypeFilterState = {
  typeLevel: ErrorTypeLevelFilter
  parentId: string
  enabled: ErrorTypeEnabledFilter
  page: number
  size: number
}

export type ErrorTypePageData = PageData<ErrorType>

File: backend/src/main/java/com/jt/learning/entity/ErrorType.java
package com.jt.learning.entity;

import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
public class ErrorType {

    private Long id;
    private Long parentId;
    private Integer typeLevel;
    private String code;
    private String name;
    private String description;
    private String nameEn;
    private String descriptionEn;
    private Integer sortOrder;
    private Boolean enabled;
    private Boolean deleted;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
